coalib/testing/LocalBearTestHelper.py

```diff
@@ -101,7 +100,7 @@ def get_results(local_bear,
         return bear_output
 
 
-class LocalBearTestHelper(unittest.TestCase):
+class LocalBearTestHelper(BaseTestHelper):
```

Comment:
this is a backwards incompatible change.

all test classes in the bears, and anywhere else that has used LocalBearTestHelper, become broken if this is merged.  They all need to be rewritten, just like LocalBearTestHelper in this patch has needed to be completely rewritten.

Most features of pytest that we want to use co-exist nicely with unittest classes.
https://github.com/pytest-dev/pytest/blob/master/doc/en/unittest.rst#pytest-features-in-unittesttestcase-subclasses

https://pypi.org/project/pytest-restrict/ explicitly supports unittest classes; that is what it is used for ... enforcing unittest classes when Django developers are using pytest, and should be using their customised Django unittest sub-classes.

Reply:
Two of the features I find most useful, namely parametrization and fixtures, do *not* work with unittest. That's why we want to get rid of it. 

But I see the difficulty of introducing this change into an existing test suite. @sangamcse do you think it's feasible to keep the unittest dependency for now and still use the base class?

Reply:
In this case, only class name has changed which is done by adding `Test` as suffix to the class name
> They all need to be rewritten, just like LocalBearTestHelper in this patch has needed to be completely rewritten.

Reply:
I tried testing coala-bears locally. All tests passed except `MarkdownBear`.
> all test classes in the bears, and anywhere else that has used LocalBearTestHelper, become broken if this is merged.

Reply:
Another idea would be to deprecate the existing `LocalBearTestHelper`, leave it as it is now and create a new one, which is based on the new base class and can be rolled out step by step.

Reply:
@jayvdb, you are right. I tried to test coala-bears by cloning this branch in another venv in my system. It collected 212 test items instead of 856 test items. So it will break coala-bears test if we go on directly with this changes.
> all test classes in the bears, and anywhere else that has used LocalBearTestHelper, become broken if this is merged. 

I agree with @NiklasMM. We can go on by creating new one, deprecating the existing `LocalBearTestHelper` if you all agree.

Reply:
@Makman2, your opinion?

Reply:
@NiklasMM , fixtures do work, with a slightly different mechanism for some types of fixtures.
see https://github.com/pytest-dev/pytest/blob/master/doc/en/unittest.rst#mixing-pytest-fixtures-into-unittesttestcase-subclasses-using-marks
and `git grep fixture tests` in coala-core repo.

wrt parameterisation, and even fixtures, I didnt see the rationale given in the cEP for how those will be used to improve bear testing.  I would argue that https://github.com/wolever/parameterized is as good a solution which works well for all test frameworks.  We already have a complex set of pytest plugins, and already use `testfixtures`, so an additional test dependency isnt a new complexity problem.

Anyway, my intention isnt to re-litigate that.  I am only commenting because of the huge breakage.

Leaving the existing unittest class as-is, and deprecating it, will reduce the risk substantially.  And then at the end of the project the huge rewrite of bear tests can happen, and it wont matter if there are some bear tests which are not migrated to pytest-only style for reasons not yet predicted.

If the immediate goal is only to enforce a common test base class, that can be achieved by changing this class to inherit from both:

```py
class LocalBearTestHelper(BaseTestHelper, unittest.TestCase):
    ...
```